jmxtrans-output/jmxtrans-output-kafka/src/main/java/com/googlecode/jmxtrans/model/output/kafka/KafkaWriterFactory.java

```diff
@@ -0,0 +1,63 @@
+package com.googlecode.jmxtrans.model.output.kafka;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.googlecode.jmxtrans.model.OutputWriterFactory;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import org.apache.kafka.clients.producer.ProducerConfig;
+import org.apache.kafka.common.serialization.StringSerializer;
+
+import javax.annotation.Nonnull;
+import java.util.Map;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+@EqualsAndHashCode(exclude = {"objectMapper", "producerConfig"})
```

Comment:
You should most probably not exclude `producerConfig` from `@EqualsAndHashCode`. To writers with different configuration are not behaving in the same way.